backend/coreapp/compiler_wrapper.py

```diff
@@ -31,34 +33,48 @@
 else:
     WINE = "wine"
 
+def load_compiler(id: str, cc: Optional[str], platform: Optional[str], compilers: Dict[str, Dict[str, str]], base_id: Optional[str]=None):
+    if not cc or not platform:
+        logger.error(f"Error: {id} {CONFIG_JSON} is missing 'cc' and/or 'platform' field(s), skipping.")
+        return
+
+    if not base_id:
+        base_id = id
+
+    # allow binaries to exist outside of repo
+    binaries_path = Path(CompilerWrapper.base_path() / base_id)
+    # consider compiler binaries present if *any* non-config.json file is found
+    binaries = (x for x in binaries_path.glob("*") if x.name != CONFIG_JSON)
+    if next(binaries, None) != None:
+        compilers[id] = {
+            "cc": cc,
+            "platform": platform
+        }
+    else:
+        logger.debug(f"Config found but no binaries found for {id}, ignoring.")
+
 def load_compilers() -> Dict[str, Dict[str, str]]:
-    ret = {}
-    config_json = "config.json"
+    ret: Dict[str, Dict[str, str]] = {}
     compilers_base = settings.BASE_DIR / "compilers"
     compiler_dirs = next(os.walk(compilers_base))
-    for compiler_id in compiler_dirs[1]:
-        config_path = Path(compilers_base / compiler_id / config_json)
+    for compiler_dir_name in compiler_dirs[1]:
+        config_path = Path(compilers_base / compiler_dir_name / CONFIG_JSON)
         if config_path.exists():
             with open(config_path) as f:
                 try:
                     config = json.load(f)
                 except:
-                    logger.error(f"Error: Unable to parse {config_json} for {compiler_id}")
+                    logger.error(f"Error: Unable to parse {CONFIG_JSON} for {compiler_dir_name}")
                     continue
 
-                if "cc" in config and "platform" in config:
-                    # allow binaries to exist outside of repo
-                    binaries_path = Path(CompilerWrapper.base_path() / compiler_id)
-                    logger.debug(f"Valid config found for {compiler_id}. Checking {binaries_path}...")
-                    # consider compiler binaries present if *any* non-config.json file is found
-                    binaries = (x for x in binaries_path.glob("*") if x.name != config_json)
-                    if next(binaries, None) != None:
-                        logger.debug(f"Enabling {compiler_id}.")
-                        ret[compiler_id] = config
-                    else:
-                        logger.debug(f"No binaries for {compiler_id}, ignoring.")
-                else:
-                    logger.warning(f"Error: {compiler_id} {config_json} is missing 'cc' and/or 'platform' field(s), skipping.")
+                if isinstance(config, dict):
+                    load_compiler(compiler_dir_name, config.get("cc"), config.get("platform"), ret)
```

Comment:
not a huge fan of mutating the dictionary being passed in - prefer the entry be returned and added to the `ret` dict
```py
cfg = load_compiler(foo, bar, baz)
if cfg is not None:
  ret.update(cfg)
```

Reply:
gonna change it